libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -63,7 +63,7 @@ private data class DynamicallyUpdatedRouteValues(
     val timeSinceLastReroute: AtomicInteger = AtomicInteger(0),
     var sessionId: String = TelemetryUtils.obtainUniversalUniqueIdentifier(),
     val distanceCompleted: AtomicReference<Float> = AtomicReference(0f),
-    val durationRemaining: AtomicLong = AtomicLong(0),
+    val durationRemaining: AtomicInteger = AtomicInteger(0),
```

Comment:
didn't see the AtomicDouble. also don't feel like we really need Atomic here 🤔

Reply:
> also don't feel like we really need Atomic here 🤔 .

@olegzil could you clarify why this was designed using Atomic?

Reply:
dynamicValues.durationRemaining is being set from multiple places in the code. It is also being read in response to telemetry events.  As long as that's the case you have to guard against the dreaded ConcurrentModificationException, which translates into an Atomic.